roles/selfsigned_certs/tasks/main.yml

```diff
@@ -0,0 +1,58 @@
+---
+- name: Install pip
+  yum:
+    name: "python2-pip"
+    state: "installed"
+  when: "'loadbalancer-vm' in group_names"
+
+- name: Install python crypto module
+  pip:
+    name: "cryptography"
+    state: "latest"
+  when: "'loadbalancer-vm' in group_names"
+
+- name: Generate a private key
+  openssl_privatekey:
+    path: "{{ tls.cert_private_path }}/{{ tls_star_cert_key }}"
+    size: 2048
+  when: "'loadbalancer-vm' in group_names"
+
+- name: Generate a CSR
+  openssl_csr:
+    path: /etc/pki/tls/certs/openconext.csr
+    privatekey_path: "{{ tls.cert_private_path }}/{{ tls_star_cert_key }}"
+    common_name: "*.{{ base_domain }}"
+    organization_name: OpenConext
+    basic_constraints: CA:TRUE
+  when: "'loadbalancer-vm' in group_names"
+
+- name: Generate a selfsigned certificate
+  openssl_certificate:
+    path: "{{ tls.cert_path }}/{{ tls_star_cert }}"
+    privatekey_path: "{{ tls.cert_private_path }}/{{ tls_star_cert_key }}"
+    csr_path: "/etc/pki/tls/certs/openconext.csr"
+    provider: selfsigned
+  when: "'loadbalancer-vm' in group_names"
+
+- name: Get the contents of the certificate
+  slurp:
+    src: "{{ tls.cert_path }}/{{ tls_star_cert }}"
+  register: selfsigned_cert
+  when: "'loadbalancer-vm' in group_names"
+  run_once: true
+
+- name: Put selfsigned cert in a variable
+  set_fact: 
+    selfsigned_cert_contents: "{{ selfsigned_cert.content | b64decode }}"
```

Comment:
What is base64 decoded and why?

Reply:
The Ansible module "slurp" is used to get the contents of that certificate into a variable, and the set_fact module is then used to make the variable usable on other hosts. The "slurp" module base64 encodes the contents of the file, hence the need to base64 decode it again